lib/autoPagination.js

```diff
@@ -0,0 +1,190 @@
+'use strict';
+
+var makeRequest = require('./makeRequest');
+var utils = require('./utils');
+
+function getItemCallback(args) {
+  if (args.length === 0) {
+    return undefined;
+  }
+  var onItem = args[0];
+  if (typeof onItem !== 'function') {
+    throw Error('The first argument to autoPagingEach, if present, must be a callback function; receieved ' + typeof onItem);
+  }
+
+  // `.autoPagingEach((item, next) => { doSomething(item); next(); });`
+  if (onItem.length === 2) {
+    return onItem;
+  }
+
+  if (onItem.length > 2) {
+    throw Error('The `onItem` callback function passed to autoPagingEach must accept at most two arguments; got ' + onItem);
+  }
+
+  // API compat; turn this:
```

Comment:
Could you add a little more flavor to this comment? What API is this trying to be compatible with?

Reply:
✅